src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -119,6 +122,9 @@ private function decideConsensus(TokenInterface $token, array $attributes, $obje
         $deny = 0;
         foreach ($this->voters as $voter) {
             $result = $voter->vote($token, $object, $attributes);
+            if (false === \is_int($result)) {
```

Comment:
what's about adding a `elseif ($result !== VoterInterface::ACCESS_ABSTAIN)` below?
It will assert the return value is a valid constant!

The deprecation message would be `return either VoterInterface::ACCESS_GRANTED, VoterInterface::ACCESS_DENIED or VoterInterface::ACCESS_ABSTAIN).

In Symfony 6.0 we will be able to type hint and replace the deprecation by an exception.